elastic/apm-server
# Sniff Content-Encoding, don't require Content-Type

## Motivation/summary

- Don't require the Content-Type header for `/intake`; assume `application/x-ndjson` if not specified.
- Sniff Content-Encoding from the request body if the header is not specified

## Checklist

~- [ ] Update [package changelog.yml](https://github.com/elastic/apm-server/blob/main/apmpackage/apm/changelog.yml) (only if changes to `apmpackage` have been made)~
~- [ ] Documentation has been updated~

## How to test these changes

Send a variety of requests to APM Server, checking that they are all accepted:
 - `Content-Encoding` unspecified, uncompressed body
 - `Content-Encoding` unspecified, gzip-compressed body
 - `Content-Encoding` unspecified, deflate-compressed body
 - `Content-Encoding: gzip`, gzip-compressed body
 - `Content-Encoding: deflate`, deflate-compressed body
 - `Content-Type` unspecified
 - `Content-Type: application/x-ndjson`

Check that these requests are rejected:
 - `Content-Type: invalid`
 - `Content-Encoding: invalid`

## Related issues

Closes https://github.com/elastic/apm-server/issues/7271
